api/index.js

```diff
@@ -492,74 +493,116 @@ router.post('/group/:groupId/leave', function(req, res) {
 
     const currentUserID = req.session.user.id;
 
-    req.db.collection(GROUP_DB).find({ id: groupID }).toArray(function(err, result) {
+    req.db.collection(GROUP_DB).find({ members: { $in: [currentUserID] }}).toArray(function(err, allGroups) {
```

Comment:
Why do we need to find all groups the user belongs to? We only need to check the expense groups inside the group the user is trying to leave...